common/org.testeditor.dsl.common.ui/src/org/testeditor/dsl/common/ui/refactoring/NamedElementRenameParticipant.xtend

```diff
@@ -0,0 +1,139 @@
+package org.testeditor.dsl.common.ui.refactoring
+
+import com.google.common.annotations.VisibleForTesting
+import java.util.Set
+import javax.inject.Inject
+import javax.inject.Named
+import org.eclipse.core.resources.IFile
+import org.eclipse.core.runtime.CoreException
+import org.eclipse.core.runtime.IProgressMonitor
+import org.eclipse.core.runtime.OperationCanceledException
+import org.eclipse.ltk.core.refactoring.Change
+import org.eclipse.ltk.core.refactoring.TextFileChange
+import org.eclipse.ltk.core.refactoring.participants.CheckConditionsContext
+import org.eclipse.ltk.core.refactoring.participants.RenameParticipant
+import org.eclipse.text.edits.ReplaceEdit
+import org.eclipse.ui.part.FileEditorInput
+import org.eclipse.xtend.lib.annotations.Accessors
+import org.eclipse.xtext.ui.editor.XtextEditor
+import org.eclipse.xtext.ui.editor.model.IXtextDocument
+import org.eclipse.xtext.ui.editor.model.XtextDocumentProvider
+import org.eclipse.xtext.ui.refactoring.impl.EditorDocumentChange
+import org.eclipse.xtext.ui.resource.IResourceSetProvider
+import org.testeditor.dsl.common.NamedElement
+import org.testeditor.dsl.common.ui.workbench.PartHelper
+
+import static org.eclipse.xtext.Constants.*
+import static org.testeditor.dsl.common.CommonPackage.Literals.*
+
+import static extension org.eclipse.xtext.nodemodel.util.NodeModelUtils.findNodesForFeature
+
+@Accessors(PROTECTED_GETTER) // for testing
+class NamedElementRenameParticipant extends RenameParticipant {
+
+	@Inject PartHelper partHelper
+	@Inject IResourceSetProvider resourceSetProvider
+	@Inject XtextDocumentProvider documentProvider
+
+	val Set<String> fileExtensions
+	@Accessors(PUBLIC_GETTER) val String name
+	IFile element
+
+	@Inject
+	new(@Named(FILE_EXTENSIONS) String fileExtensions, @Named(LANGUAGE_NAME) String languageName) {
+		this.fileExtensions = fileExtensions.split(",").map[toLowerCase].toSet
```

Comment:
We should add a test-case if the rename participant behaves properly if a file like `MyTest.TCL` is passed.